Assets/Tests/Runtime/ConnectionManagementTests.cs

```diff
@@ -0,0 +1,514 @@
+                builder.RegisterComponent(ConnectionManager);
+                builder.RegisterComponent(UpdateRunner);
+                builder.RegisterComponent(new NetworkedMessageChannel<ConnectionEventMessage>()).AsImplementedInterfaces();
+                builder.RegisterInstance(new BufferedMessageChannel<ConnectStatus>()).AsImplementedInterfaces();
+                builder.RegisterInstance(new MessageChannel<UnityServiceErrorMessage>()).AsImplementedInterfaces();
+                builder.RegisterInstance(new MessageChannel<ReconnectMessage>()).AsImplementedInterfaces();
+                builder.RegisterInstance(new BufferedMessageChannel<LobbyListFetchedMessage>()).AsImplementedInterfaces();
+                builder.Register<LocalLobby>(Lifetime.Singleton);
+                builder.Register<LocalLobbyUser>(Lifetime.Singleton);
+                builder.Register<ProfileManager>(Lifetime.Singleton);
+                builder.RegisterEntryPoint<LobbyServiceFacade>(Lifetime.Singleton).AsSelf();
+            }
+        }
+
+        class SceneLoaderWrapperStub : SceneLoaderWrapper
+        {
+            public override void Awake()
+            {
+                Instance = this;
+            }
+
+            public override void Start() { }
+
+            public override void AddOnSceneEventCallback() { }
+
+            public override void LoadScene(string sceneName, bool useNetworkSceneManager, LoadSceneMode loadSceneMode = LoadSceneMode.Single) { }
+        }
+
+        protected override int NumberOfClients => 2;
+
+        ConnectionManagementTestsLifeTimeScope[] m_ClientScopes;
+        ConnectionManagementTestsLifeTimeScope m_ServerScope;
+
+        ConnectionManager[] m_ClientConnectionManagers;
+        ConnectionManager m_ServerConnectionManager;
+
+
+        protected override bool CanStartServerAndClients()
+        {
+            return false;
+        }
+
+        static void InitializeInstance(string name, NetworkManager networkManager, out ConnectionManager connectionManager, out ConnectionManagementTestsLifeTimeScope scope)
+        {
+            var connectionManagerGO = new GameObject($"ConnectionManager - {name}");
+            connectionManager = connectionManagerGO.AddComponent<ConnectionManager>();
+
+            networkManager.NetworkConfig.ConnectionApproval = true;
+            networkManager.NetworkConfig.EnableSceneManagement = false;
+
+            var updateRunnerGO = new GameObject($"UpdateRunner - {name}");
+            var updateRunner = updateRunnerGO.AddComponent<UpdateRunner>();
+
+            var lifeTimeScopeGO = new GameObject($"LifeTimeScope - {name}");
+            scope = lifeTimeScopeGO.AddComponent<ConnectionManagementTestsLifeTimeScope>();
+            scope.NetworkManager = networkManager;
+            scope.ConnectionManager = connectionManager;
+            scope.UpdateRunner = updateRunner;
+            scope.Build();
+        }
+
+        void CreatePlayerPrefab()
```

Comment:
From offline discussion, check if this is still needed.

Reply:
after double-checking, it is still needed. Made the reason clearer in the comments